### PR TITLE
get files without uploading nanoglob.py

### PR DESCRIPTION
**Pull request by dsl400**

### CODE REVIEW

**Comment by dsl400** [comment]
not sure if it's 100% functional ....

File: upydev/shell/shfileio.py [hunk 1]
```diff
@@ -366,10 +379,18 @@ def fileop(self, cmd, args, rest_args):
                 except Exception:
                     return
             print('get: searching files...')
```
Comment (Carglglz) [inline upydev/shell/shfileio.py:381]: use `fg` arg as:
```python
if not args.fg:
   file_tree = self.read_file_tree()
   for patt in rest_args:
                 if '*' not in patt: # not pattern searching
                     for f in file_tree:
                         if f != patt: continue
                         file_match.append((file_tree[f],f))
                 else:               # pattern searching
                     pattrn = re.compile(patt.replace('.', r'\.').replace('*', '.*') + '$')
                     for f in file_tree:
                         if not pattrn.match(file): continue
                         file_match.append((file_tree[f],f))
else:
    < current method >
```

**Comment by Carglglz** [comment]
I've just tested it:

Filenames OK
but with wildcards i.e. pattern matching 
```
$ upydev get "ma*.py" -fg -@ esp8266
get: searching files...
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/bin/upydev", line 1038, in <module>
    handle_action(args)
  File "/Library/Frameworks/Python.framework/Versions/3.7/bin/upydev", line 461, in handle_action
    device=dev_name)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/upydev/fileio.py", line 234, in fileio_action
    sh.cmd(cmd_inp)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/upydev/shell/commands.py", line 205, in cmd
    self.sh_cmd(cmdline)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/upydev/shell/commands.py", line 1391, in sh_cmd
    self.custom_sh_cmd(command, rest_args, args, self.topargs, unknown_args)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/upydev/shell/shserial.py", line 310, in custom_sh_cmd
    self.dsyncio.fileop(cmd, args, rest_args)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/upydev/shell/shfileio.py", line 401, in fileop
    if not pattrn.match(file): continue
UnboundLocalError: local variable 'file' referenced before assignment
```

**Comment by Carglglz** [comment]
@dsl400 
Thanks for your contribution, this will be in the next release 👍🏼

**Pull request merged**